tests/algorithm/binarySearchTree.test.ts

```diff
@@ -0,0 +1,22 @@
+/// <reference types="jest" />
+import { binarySearchTree } from "../../src/algorithm/binarySearchTree"
+
+test("insertion", () => {
+	var bst = new binarySearchTree<string, number>()
+	var input = "S E A R C H E X A M P L E".split(' ')
```

Comment:
By the way, I think kinda like `Set`, Keys in a BST are supposed to be distinct.
Shouldn't that code throw an error with multiple `E` being used as keys and overwriting each other?

Reply:
Yes, So If the key already exsits, the value would be updated. 👍   
Sure I can add the `Has` function!

Reply:
Ha ok, indeed that make sense too, so I think it's just a naming issue that make it a bit confusing to me.

For me `.insert()` is a kind of synonym of `.add()` and implies that something new is added and does not overwrite previous data, it throw an error or do nothing.

Maybe we should use `.set()` like the [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) structure to make this behaviour more explicit

Reply:
Yep! I will add them!

Reply:
Ha ok, indeed that make sense too, so I think it's just a naming issue that make it a bit confusing to me.

For me `.insert()` is a kind of synonym of `.add()` and implies that something new is added and does not overwrite previous data, it throw an error or do nothing.

Maybe we should use `.set()` like the [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) structure to make this behaviour more explicit

Reply:
Ha ok, indeed that make sense too, so I think it's just a naming issue that make it a bit confusing to me.

For me `.insert()` is a kind of synonym of `.add()` and implies that something new is added and does not overwrite previous data, it throw an error or do nothing.

Maybe we should use `.set()` like the [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) structure to make this behaviour more explicit